app/src/main/java/com/demo/mdb/pokedex/MainActivity.java

```diff
@@ -0,0 +1,154 @@
+package com.demo.mdb.pokedex;
+
+import android.os.Bundle;
+import android.support.annotation.BoolRes;
+import android.support.design.widget.FloatingActionButton;
+import android.support.design.widget.Snackbar;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.Toolbar;
+import android.view.View;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.Button;
+import android.widget.CompoundButton;
+import android.widget.Switch;
+import android.content.Intent;
+
+import java.util.ArrayList;
+
+public class MainActivity extends AppCompatActivity {
+
+    Button button;
+    Switch lowSwitch;
+    Switch medSwitch;
+    Switch highSwitch;
+    Boolean low = true;
+    Boolean med = true;
+    Boolean high = true;
+    Pokedex pokedex = new Pokedex();
+    Boolean isGridView = false;
+    final ArrayList<Pokedex.Pokemon> pokemonArrayList = pokedex.getPokemon();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        lowSwitch = (Switch) findViewById(R.id.lowSwitch);
+        medSwitch = (Switch) findViewById(R.id.medSwitch);
+        highSwitch = (Switch) findViewById(R.id.highSwitch);
+        button = (Button) findViewById(R.id.button);
+
+        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
+        fab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent intent = new Intent(MainActivity.this, SearchActivity.class);
+                intent.putExtra("isGridView", isGridView);
+                startActivity(intent);
+            }
+        });
+
+        final RecyclerView recyclerView = (RecyclerView) findViewById(R.id.recyclerView);
+        recyclerView.setLayoutManager(new LinearLayoutManager(this));
+
+        final ListViewAdapter listPokemonAdapter = new ListViewAdapter(getApplicationContext(), pokemonArrayList);
+        final GridViewAdapter gridPokemonAdapter = new GridViewAdapter(getApplicationContext(), pokemonArrayList);
+        recyclerView.setAdapter(listPokemonAdapter);
+
+        lowSwitch.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
+                low = b;
+                listPokemonAdapter.pokemonArrayList = updatedArrayList();
```

Comment:
Do you really need to call it twice back to back, or can you just save the results?